contracts/rsv/Reserve.sol

```diff
@@ -0,0 +1,367 @@
+
+    function changeFeeRecipient(address newFeeRecipient) external onlyOwnerOr(feeRecipient) {
+        feeRecipient = newFeeRecipient;
+        emit FeeRecipientChanged(newFeeRecipient);
+    }
+
+    /// Make a different address own the EternalStorage contract.
+    /// This will break this contract, so only do it if you're
+    /// abandoning this contract, e.g., for an upgrade.
+    function transferEternalStorage(address newOwner) external onlyOwner {
+        data.transferOwnership(newOwner);
+    }
+
+    /// Change the contract that helps with transaction fee calculation. 
+    function changeTxFeeHelper(address newTxFee) external onlyOwner {
+        txFee = ITXFee(newTxFee);
+    }
+
+    /// Change the maximum supply allowed.
+    function changeMaxSupply(uint256 newMaxSupply) external onlyOwner {
+        maxSupply = newMaxSupply;
+        emit MaxSupplyChanged(newMaxSupply);
+    }
+
+    /// Change the name and ticker symbol of this token.
+    function changeName(string calldata newName, string calldata newSymbol) external onlyOwner {
+        name = newName;
+        symbol = newSymbol;
+        emit NameChanged(newName, newSymbol);
+    }
+
+    /// Pause the contract.
+    function pause() external only(pauser) {
+        paused = true;
+        emit Paused(pauser);
+    }
+
+    /// Unpause the contract.
+    function unpause() external only(pauser) {
+        paused = false;
+        emit Unpaused(pauser);
+    }
+
+    /// Modifies a function to run only when the contract is not paused.
+    modifier notPaused() {
+        require(!paused, "contract is paused");
+        _;
+    }
+
+    /// Freeze token transactions for a particular address.
+    function freeze(address account) external only(freezer) {
+        require(data.frozenTime(account) == 0, "account already frozen");
+
+        // In `wipe` we use block.timestamp (aka `now`) to check that enough time has passed since
+        // this freeze happened. That required time delay -- 4 weeks -- is a long time relative to
+        // the maximum drift of block.timestamp, so it is fine to trust the miner here.
+        // solium-disable-next-line security/no-block-members
+        data.setFrozenTime(account, now);
+
+        emit Frozen(freezer, account);
+    }
+
+    /// Unfreeze token transactions for a particular address.
+    function unfreeze(address account) external only(freezer) {
+        require(data.frozenTime(account) > 0, "account not frozen");
+        data.setFrozenTime(account, 0);
+        emit Unfrozen(freezer, account);
+    }
+
+    /// Modifies a function to run only when the `account` is not frozen.
+    modifier notFrozen(address account) {
+        require(data.frozenTime(account) == 0, "account frozen");
+        _;
+    }
+
+    /// Burn the balance of an account that has been frozen for at least 4 weeks.
+    function wipe(address account) external only(freezer) {
+        require(data.frozenTime(account) > 0, "cannot wipe unfrozen account");
```

Comment:
Why check this when also checking that it's been frozen for at least 4 weeks? Seems like a subset of the later check.

Reply:
Wiping is gone in the new version